packages/react-utils/src/rewriteReactNodeRecursively.ts

```diff
@@ -0,0 +1,37 @@
+import { ReactNode, isValidElement, cloneElement, Key } from 'react';
+
+type RewriteReactNodeRecursivelyCallback = (stringSubNode: string) => string;
+
+/**
+ * Recursively crawls ReactNode and applies callback on all string subnodes
+ * @param node ReactNode to be recursively crawled & modified
+ * @param callback function to be applied on all string subnodes
+ * @param reactKey optional React key applied on the root node, if it is a ReactElement. Used internally.
+ * @returns modified ReactNode
+ */
+export const rewriteReactNodeRecursively = (
+    node: ReactNode,
+    callback: RewriteReactNodeRecursivelyCallback,
+    reactKey?: Key,
+): ReactNode => {
+    if (typeof node === 'string') return callback(node);
+    if (typeof node === 'number') return callback(node.toString());
+
+    if (Array.isArray(node)) {
+        return node.map((child, index) => {
+            const newReactKey: Key | undefined = isValidElement(child)
+                ? child.key ?? index
+                : undefined;
+
+            return rewriteReactNodeRecursively(child, callback, newReactKey);
+        });
+    }
+
+    if (isValidElement(node)) {
+        const rewrittenChildren = rewriteReactNodeRecursively(node.props.children, callback);
+
+        return cloneElement(node, { ...node.props, key: reactKey, children: rewrittenChildren });
```

Comment:
**TL;DR: Should I scrap this?** :thinking: 

My idea was to have `HiddenPlaceholder.tsx` alone handle redacting numbers, so you would just wrap any content with it and it'd be completely automatic.
As I wrote in the comment there, this only works for simple cases of components wrapping children, such as:
`<Wrapper>Label: <SomeStyle>12345 CZK</SomeStyle></Wrapper>`.

But this procedure simply stops when it encounters more complex components, which calculate their content instead of just wrapping children, like  `<Formatter value={1234} />`. They're returned "as is".

So those components have to do the redacting themselves: see `prepareFiatAmountFormatter.ts`, `prepareCryptoAmountFormatter.ts`.

I'm not happy with this mixed approach, where _sometimes_ it does things automagically under the hood, but _sometimes_ you just have to do it yourself. I fear this might lead to confusion and mistakes later on.
So should we drop this magic, and redact numbers at all specific places where we display them?
Reminder: this won't make `HiddenPlaceholder.tsx` useless, it still does CSS blur, and will provide the context for `shouldRedactNumbers`.